functions/Test-DbaSqlPath.ps1

```diff
@@ -1,96 +1,99 @@
+			The Path to test. This can be a file or directory
+
+		.PARAMETER Silent
+			If this switch is enabled, the internal messaging functions will be silenced.
+
+		.OUTPUTS
+			System.Boolean
+
+		.NOTES
+			Tags: Path, ServiceAccount
+			Author: Chrissy LeMaire (@cl), netnerds.net
+			Requires: Admin access to server (not SQL Services),
+			Remoting must be enabled and accessible if $SqlInstance is not local
+
+			dbatools PowerShell module (https://dbatools.io, clemaire@gmail.com)
+			Copyright (C) 2016 Chrissy LeMaire
+			License: GNU GPL v3 https://opensource.org/licenses/GPL-3.0
+
+		.LINK
+			https://dbatools.io/Test-DbaSqlPath
+
+		.EXAMPLE
+			Test-DbaSqlPath -SqlInstance sqlcluster -Path L:\MSAS12.MSSQLSERVER\OLAP
+
+			Tests whether the service account running the "sqlcluster" SQL Server instance can access L:\MSAS12.MSSQLSERVER\OLAP. Logs into sqlcluster using Windows credentials.
+
+		.EXAMPLE
+			$credential = Get-Credential
+			Test-DbaSqlPath -SqlInstance sqlcluster -SqlCredential $credential -Path L:\MSAS12.MSSQLSERVER\OLAP
+
+			Tests whether the service account running the "sqlcluster" SQL Server instance can access L:\MSAS12.MSSQLSERVER\OLAP. Logs into sqlcluster using SQL authentication.
+
+	#>
 	[CmdletBinding()]
 	[OutputType([bool])]
 	param (
 		[Parameter(Mandatory = $true)]
 		[Alias("ServerInstance", "SqlServer")]
-		[DbaInstanceParameter]
-		$SqlInstance,
+		[DbaInstance[]]$SqlInstance,
+		[PSCredential]$SqlCredential,
 		[Parameter(Mandatory = $true)]
 		[string]$Path,
-		[System.Management.Automation.PSCredential]
-		$SqlCredential,
 		[switch]$Silent
 	)
-    
-	try {
-		$server = Connect-SqlInstance -SqlInstance $SqlInstance -SqlCredential $SqlCredential
-	}
-	catch {
-		Stop-Function -Message "Failure" -Category ConnectionError -ErrorRecord $_ -Target $instance -Continue
-		return
-	}
-    
-	Write-Message -Level VeryVerbose -Message "Path check is $path."
-	$sql = "EXEC master.dbo.xp_fileexist '$path'"
-	try {
-		Write-Message -Level Debug -Message "Executing: $sql."
-		$fileexist = $server.ConnectionContext.ExecuteWithResults($sql)
-	}
-    
-	catch {
-		Stop-Function -Message "Failed to test the path $Path." -ErrorRecord $_ -Target $SqlInstance
-		return
-	}
-	if ($fileexist.tables.rows[0] -eq $true -or $fileexist.tables.rows[1] -eq $true) {
-		return $true
+
+	process {
+		foreach ($instance in $SqlInstance) {
+			try {
+				$server = Connect-SqlInstance -SqlInstance $instance -SqlCredential $SqlCredential
+			}
+			catch {
+				Stop-Function -Message "Failure" -Category ConnectionError -ErrorRecord $_ -Target $instance -Continue
+			}
+
+			Write-Message -Level VeryVerbose -Message "Checking acces to $path for $instance."
+			$sql = "EXEC master.dbo.xp_fileexist '$path'"
+			try {
+				Write-Message -Level Debug -Message "Executing: $sql."
+				$fileExist = $server.Query($sql)
+			}
+
+			catch {
+				Stop-Function -Message "Failed to test the path $Path." -ErrorRecord $_ -Target $SqlInstance
+				return
```

Comment:
The target should be `$instance` and you'd need to use the `-Continue` parameter here, instead of using return on the next line

Reply:
Done. Caught this same issue on other commands.